packages/gotrue/lib/src/gotrue_client.dart

```diff
@@ -581,7 +581,7 @@ class GoTrueClient {
 
   /// Force refreshes the session including the user data in case it was updated
   /// in a different session.
-  Future<AuthResponse> refreshSession() async {
+  Future<AuthResponse?> refreshSession() async {
```

Comment:
This would be a breaking change, and we cannot do it. Same with `setSession` and `recoverSession`.

Reply:
Could we expose a tryRefreshSession, that refreshSession then wraps, and then keeps the current method of throwing an exception?